vortex-array/src/array/chunked/canonical.rs

```diff
@@ -166,13 +166,13 @@ fn swizzle_struct_chunks(
 fn pack_bools(chunks: &[Array], nullability: Nullability) -> VortexResult<BoolArray> {
     let len = chunks.iter().map(|chunk| chunk.len()).sum();
     let validity = validity_from_chunks(chunks, nullability);
-    let mut bools = Vec::with_capacity(len);
+    let mut buffer = BooleanBufferBuilder::new(len);
```

Comment:
building BooleanBuffer avoids copy compared to creating BooleanBuffer from Vec<bool>